# ZIKV_diagnostic-
Data and scripts from the manuscript:  "Identification of diagnostic peptide regions that distinguish Zikva virus from related mosquito-borne Flaviviruses"

Description of files

1.  Sequence data:  
	Unaligned FASTA sequence files for each of the 10 species can be found in directories:
	/Sequence_data/E_seq/ and /Sequence_data/NS1_seq

	MSA alignment across all 10 species can be found in files:
	/Sequence_data/E_seq/Flavi_E_combined.fasta and /Sequence_data/NS1_seq/Flavi_NS1_combined.fasta

	The GB accessions for all the sequences can be found in Supplementary Table 1:  /Sequence_data/S1_Table.xlsx
	
	The Meta-CATS tool, developed by the Virus Pathogen Resource (ViPR; www.viprbrc.org) and Influenza Research Database (IRD; www.fludb.org) teams, was used to compute the variation present at each column of the alignment. 

	Code for the Meta-CATS analysis can be found in directory:
	/Sequence_data/Meta-CATS_Variation/metadata_parser.pl

	The metadata_parser.pl file is a wrapper script that parses the input files (taxa assignment and sequence alignment files) and passes them to the meta-CATS script (chisq_test.R) which outputs several statistics files.  
	Note: we are only using the residue counts for each of the (species vs. all-other-Flavi's) for downstream analyses.
	
	Taxa assignment files can be found in /Sequence_Data/E_seq/Flavi_E_Taxa_Assignments.tsv OR /Sequence_Data/NS1_seq/Flavi_NS1_Taxa_Assignments.tsv
	Alignment files are located in /Sequence_Data/E_seq/Flavi_E_combined.fasta.afa OR /Sequence_Data/NS1_seq/Flavi_NS1_combined.fasta.afa
	
	Change lines 15 & 16 of the "metadata_parser.pl" Perl script to the appropriate files containing the taxa assignments metadata and aligned sequences respectively.

	Once the script finishes, the modified alignment files and residue distributions will be reported in separate output files for each taxon. A subset of these files will be used later in this workflow (see #2).

2.  Sensitivity and specificity:

	Reformated sequence data files can be found in directory:
	/Sequence_data_reformat/

	Run /Sensitivity_Specificity/sensitivity_specificity_MCbatch.py
		Change directory inputs (lines 27 and 28) to be directory of reformated E and NS1 sequence data
		Output can be found in directories: /Sensitivity_Specificity/E/ and /Sensitivity_Specificity/NS1/

3.  Diagnostic sites:
	Each of the excel files containing the sensitivity and specificity scores (from step 2) were aggregated into Supplementary Table 2 and Table 3 for the E and NS1 proteins respectively.  Within the excel files, each sheet corresponds to one of the species:
	/Sensitivity_specificity/E/S2_Table.xlsx and /Sensitivity_specificity/NS1/S3_Table.xlsx

	In S2_Table.xlsx:
		For each sheet in the file:
			Copy columns A through I from /Sensitivity_specificity/E/<virus name>_E_stats.xlsx and paste into S2_Table.xlsx
			Add column called "Average": {=average(Sensitivity, Spcificity)}
			Add column called "if average>=0.98": {=if(Average>=0.98,1,0)}
			Add column called "numDiagnostic": {=sum(if avg>=0.98)} (Note:  Not shown, but counts can be found in Table 3 in text)
	Similarly for S3_Table.xlsx

4.  Diagnostic peptides:
	Excel files containing the number of diagnostic peptides can be found in files:
	/Diagnostic_peptides/Count_diagnostic_sites_in_15mer_E.xlsx and /Diagnostic_peptides/Count_diagnostic_sites_in_15mer_NS1.xlsx

	These excel files were generated by copying and pasting the "site" column and "if average>=0.98" column from
	/Sensitivity_specificity/E/S2_Table.xlsx and /Sensitivity_specificity/NS1/S3_Table.xlsx

	into "Site" and "<virus name>_ifDiagnostic" columns of /Diagnostic_peptides/Count_diagnostic_sites_in_15mer_E.xlsx and /Diagnostic_peptides/Count_diagnostic_sites_in_15mer_NS1.xlsx

	In Count_diagnostic_sites_in_15mer_E.xlsx:
		For each sheet in file:
			Add column called "<virus name>_numOfDiagnostics":  number of sites that are diagnostic within a 15mer
			Add column called "<virus name>_3orMore":  number of 15mers that have at least 3 diagnostic sites

	Similarly for Count_diagnostic_sites_in_15mer_NS1.xlsx

5.  Surface accessability:
	In /Surface_accessability/areaSAS_E_monomer.xlsx:
		In sheet "areaSAS":
			"Position in MSA":  position based on alignment of sequences from all 10 Flavivirus taxa
			"AA residue":  amino acid residue on the ZIKV protein 5IRE
			"Position in Chimera":  position based on the ZIKV 5IRE protein structure
			"transmembrane annotation": 1 if position in MSA is located in transmembrane region, based on uniprot annotation (http://www.uniprot.org/uniprot/A0A060H177)
			"adjusted areaSAS":  areaSAS for all position, 0 if position if located in transmembrane region
		In sheet "RSA":
			"Position in MSA":  position based on alignment of sequences from all 10 Flavivirus taxa
			Columns B through U calculate the RSA value at each position = areaSAS normalized by the maximum area of the the residue at that position (see sheet maxArea_perResidue)
			"RSA":  normalized surface area
			"Average hydrophilic residue values":  average of RSA scores that had hydrophilic residue at that position
			"if RSA>= 0.321":  1 if RSA score exceeded the average hydrophilic value
			"Num surf exposed in 15-mer": number of sites within a 15mer that had RSA score that exceeded the average hydrophilic score
	Similarly for areaSAS_NS1_dimer.xlsx, except there are 2 areaSAS values from each chain of the NS1 protein and the minimum value between the two was used in the RSA calculation

	6.  Surface diagnostic peptides:
		Excel files containing the number of surface diagnostic peptides can be found in files:
		/Overlay_surf_diagnostic/Surf_overlay_diag_E.xlsx and /Overlay_surf_diagnostic/Surf_overlay_diag_NS1.xlsx

		In Surf_overlay_diag_E.xlsx:
			For each sheet in file:
				Copy "<virus name>_numOfDiagnostics" column from "Count_diagnostic_sites_in_15mer_E.xlsx" (from #4) and paste into Surf_overlay_diag_E.xlsx
				Copy "Num of surf exposed in 15mer" column from "areaSAS_E_monomer.xlsx" and paste into "Surf_overlay_diag_E.xlsx"
				Add column called "Number of diagnostic sites in 15-mer (overlap)": is the number of diagnostic sites in the 15mer if the 15mer contains at least 3 sites and the number of surface exposed sites is at least 6
				Add column called "Number of diagnostic sites in 15-mer": is the number of diagnostic sites in the 15mer if the 15mer contains less than 3 sites or the number of surface exposed sites is less than 6
				Add column called "Number of surface exposed sites in 15-mer (overlap)": is the number of surface sites in the 15mer if the 15mer contains at least 3 sites and the number of surface exposed sites is at least 6
				Add column called "Number of surface exposed sites in 15-mer": is the number of surface sites in the 15mer if the 15mer contains less than 3 sites or the number of surface exposed sites is less than 6
				*These columns are used to create the bar plot 
		Similarly for Surf_overlay_diag_NS1.xlsx and Surf_overlay_diag_allTaxa (allTaxa looks for shared surface diagnostic peptides that have at least 1 diagnostic site and at least surface exposed sites)


	7.  3D protein structure:
		Coloring of the 3D protein structures can be found in /3D_protein_overlay/Manual_color_surfExp_sites.xlsx

		In sheet "ZIKV_E":
			"Position in MSA":  position based on alignment of sequences from all 10 Flavivirus taxa
			"AA residue":  amino acid residue on the ZIKV protein 5IRE
			"Position in 5IRE":  position based on the ZIKV 5IRE protein structure
			"if RSA >= 0.321":  1 if site is surface exposed (from #5)
			"Selected 15mer":  1 if position in 15mer we highlighted in the paper
			"ZIKV_ifDiagnostic":  1 if site is diagnostic (from #3)
			"Positions in 5IRE that are surf exposed (Blue)":  Sites that are surface exposed are colored blue in Figure 4
			"Positions in selected 15mer peptide (Red)":  Sites that located in selected 15mer peptide are colored red in Figure 4 
			"Positions that overlap surf exposed, 15mer (Purple)":  Sites that are surface exposed and are located in the selected 15mer are colored in purple in Figure 4
			"Positions that overlap surf exposed, diagnostic (Pink)":  Sites that are surface exposed and diagnostic within the selected 15mer are colored pink in Figure 4
		Similarly in "ZIKV_NS1" sheet.

		Chimera sessions can be uploaded using /3D_protein/overlay/ZIKV_E_overlay_selected15mer.py and /3D_protein/overlay/ZIKV_NS1_overlay_selected15mer.py


	8.  B-cell overlap:
		Run /Bcell_overlap/Overlap_E_Bcell_epitopes.py to calculate the percentage of sites that are within surface diagnostic 15mer peptides and also located within known B-cell epitopes

		Input:
			MSA alignment of E protein across all taxa
			Continuous E B-cell epitopes found in ViPR
			Discontinuous E B-cell epitopes found in ViPR (these were manually inspected to make sure the coordinates matched the alignment file)
			Excel file of the number of diagnostic sites and surface exposed sites for each Flavivirus species
			Name of the output file

		Output
			File containing 1/0 if site is located within B-cell epitope
			Counts for the number of sites within surface diagnostic peptides (printed to screen)
			Counts for the number of sites within surace diagnostic peptides and B-cell epitopes (printed to screen)

